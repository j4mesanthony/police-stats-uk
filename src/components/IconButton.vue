<script setup lang="ts">
const props = defineProps({
    icon: {
        type: String,
        required: true,
        validator: (icon) => icon !== ''
    },

    showOutline: {
        type: Boolean,
        default: true
    }
});
</script>

<template>
    <div class="p-2 top-[7px] right-[7px] rounded-md h-[38px] hover:shadow-md box-border text-center hover:border-cyan-500 border-[1px] border-slate-900 cursor-pointer duration-100 bg-slate-950 w-fit group">
        <span class="material-symbols-outlined relative text-[20px] transition-transform  text-slate-400 group-hover:text-cyan-500 group-hover:animate-pulse group-hover:scale-[1.3]">{{ props.icon }}</span>
    </div>
</template>