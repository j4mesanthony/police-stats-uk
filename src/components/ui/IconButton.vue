<script setup lang="ts">
const props = defineProps({
    icon: {
        type: String,
        required: true,
        validator: (icon) => icon !== ''
    },

    showOutline: {
        type: Boolean,
        default: true
    }
});
</script>

<template>
    <div :class="{ 'hover:shadow-md bg-slate-950 hover:border-cyan-500 border-[1px] border-slate-900': props.showOutline }" class="p-2 top-[7px] right-[7px] rounded-md h-[38px] box-border text-center select-none cursor-pointer duration-100 w-fit group">
        <span data-test="icon" class="material-symbols-outlined relative text-[20px] transition-transform  text-slate-400 group-hover:text-cyan-500 group-hover:animate-pulse group-hover:scale-[1.2]">{{ props.icon }}</span>
    </div>
</template>