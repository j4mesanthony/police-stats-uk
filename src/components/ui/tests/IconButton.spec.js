import { describe, it, expect } from 'vitest';
import IconButton from '../IconButton.vue';
import { createApp } from 'vue';
import { renderToString } from '@vue/server-renderer';

describe('IconButton', () => {
    it('should not regress (snapshot test)', async () => {
        const app = createApp(IconButton);
        const string = await renderToString(app);
        expect(string).toMatchSnapshot();
    })
})