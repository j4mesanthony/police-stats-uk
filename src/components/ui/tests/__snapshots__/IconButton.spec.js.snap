// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconButton > should not regress (snapshot test) 1`] = `"<div class=\\"hover:shadow-md bg-slate-950 hover:border-cyan-500 border-[1px] border-slate-900 p-2 top-[7px] right-[7px] rounded-md h-[38px] box-border text-center select-none cursor-pointer duration-100 w-fit group\\"><span class=\\"material-symbols-outlined relative text-[20px] transition-transform text-slate-400 group-hover:text-cyan-500 group-hover:animate-pulse group-hover:scale-[1.2]\\"></span></div>"`;
